test/createAzureResource.test.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import * as fse from 'fs-extra';
+import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, DialogResponses, IAzureNode, IAzureTreeItem, TestAzureAccount, TestUserInput } from 'vscode-azureextensionui';
+import { ext } from '../src/extensionVariables';
+import { FunctionAppProvider } from '../src/tree/FunctionAppProvider';
+import { FunctionAppTreeItem } from '../src/tree/FunctionAppTreeItem';
+import * as fsUtil from '../src/utils/fs';
+
+// This ensures that this will only run on nightly cron builds
+if (process.env.TRAVIS_EVENT_TYPE === 'cron') {
+    suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Promise<void> {
+        this.timeout(1200 * 1000);
+        const appName: string = 'msftazurefunctionscti';
+        const testFolderPath: string = path.join(os.tmpdir(), `azFunc.createNewProjectTests${fsUtil.getRandomHexString()}`);
+
+        suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+            this.timeout(120 * 1000);
+            const outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel('Azure Functions Test');
+            ext.outputChannel = outputChannel;
+            const testAccount: TestAzureAccount = new TestAzureAccount();
+            await testAccount.getTestSubscription();
+            ext.tree = new AzureTreeDataProvider(new FunctionAppProvider(ext.outputChannel), 'azureFunctions.startTesting', undefined, testAccount);
+            await fse.ensureDir(testFolderPath);
+        });
+
+        suiteTeardown(async () => {
+            ext.ui = new TestUserInput([appName]);
+            const subscriptionNode: IAzureNode<IAzureTreeItem> = (await ext.tree.getChildren())[0];
```

Comment:
Yeah, makes sense for the clean-up method so we don't rely on something that can break.  Though I think I would want to keep it that way for the actual tests since it's a good way to validate the end-to-end process of creating an app is functional.